android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java

```diff
@@ -369,7 +369,11 @@ public void onReceiveValue(String value) {
     }
 
     public void onMessage(String message) {
-      dispatchEvent(this, new TopMessageEvent(this.getId(), message));
+      if (mRNCWebViewClient != null) {
+        WritableMap eventData = mRNCWebViewClient.createWebViewEvent(this, this.getUrl());
+        eventData.putString("data", message);
+        dispatchEvent(this, new TopMessageEvent(this.getId(), eventData));
```

Comment:
Do we want this line outside of the conditional? If `mRNCWebViewClient` is null, it won't get dispatched.

Reply:
anyway to guarantee `mRNCWebViewClient` is not `null`? from the previous code, it seems that `mRNCWebViewClient` wont be `null`

Reply:
@jl9404 If `mRNCWebViewClient` is null, then do you not want to dispatch that event, then? I thought lines 373-374 were the only ones that needed to check if it was null, but I could be wrong here.

Reply:
mRNCWebViewClient has not reason to be null at this point in the code, and I feel like I'd rather miss an event than crash if it happened to be null ^^'

Reply:
Not to belabor the point, but I was wondering if we wanted to do this, which would both protect against null and not miss an event:

```js
    public void onMessage(String message) {
      if (mRNCWebViewClient != null) {
        WritableMap eventData = mRNCWebViewClient.createWebViewEvent(this, this.getUrl());
        eventData.putString("data", message);
      }
      dispatchEvent(this, new TopMessageEvent(this.getId(), eventData));
    }
```

Reply:
@jamonholmgren you can't access eventData in the code your wrote

Reply:
Oops, you're totally right. 🤦‍♂️

Reply:
@Titozzz @jamonholmgren Maybe
```js
public void onMessage(String message) {
  WritableMap eventData;
  if (mRNCWebViewClient != null) {
    eventData = mRNCWebViewClient.createWebViewEvent(this, this.getUrl());
  } else {
    eventData = Arguments.createMap();
  }
  eventData.putString("data", message);
  dispatchEvent(this, new TopMessageEvent(this.getId(), eventData));
}
```
?

Reply:
BTW `this.getURL()` throws
`java.lang.Throwable: A WebView method was called on thread 'JavaBridge'. All WebView methods must be called on the same thread. (Expected Looper Looper (main, tid 2) {eead3fd} called on Looper (JavaBridge, tid 1541) {63b1022}, FYI main Looper is Looper (main, tid 2) {eead3fd})`

Reply:
Working code:
```js
public void onMessage(String message) {
  if (mRNCWebViewClient != null) {
    WebView wv = this;
    wv.post(new Runnable() {
      @Override
      public void run() {
        WritableMap data = mRNCWebViewClient.createWebViewEvent(wv, wv.getUrl());
        data.putString("data", message);
        dispatchEvent(wv, new TopMessageEvent(wv.getId(), data));
      }
    });
  } else {
    WritableMap eventData = Arguments.createMap();
    eventData.putString("data", message);
    dispatchEvent(this, new TopMessageEvent(this.getId(), eventData));
  }
}
```
But all data inside `data` are empty in my case.